src/App.tsx

```diff
@@ -1,223 +1,75 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import Error from './components/Error/Error';
+import { ErrorContext } from './components/Error/ErrorContext';
+import { TodoContext } from './components/Todo/TodoContext';
+import TodoFooter from './components/Todo/TodoFooter/TodoFooter';
+import TodoForm from './components/Todo/TodoForm';
+import TodoList from './components/Todo/TodoList/TodoList';
+import { ErrorContextType } from './types/ErrorContextType';
+import { TodoContextType } from './types/TodoContextType';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const {
+    todos,
+    setTodos,
+    visibleTodos,
+  } = useContext(TodoContext) as TodoContextType;
+  const { setIsError, setErrorText }
+  = useContext(ErrorContext) as ErrorContextType;
+  const [isAdding, setIsAdding] = useState(false);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const getTodosFromServer = async () => {
+    if (!user) {
+      return;
+    }
+
+    try {
+      const result = await getTodos(user.id);
+
+      setTodos(result);
+    } catch {
+      setIsError(true);
+      setErrorText('Unable to load todos');
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, [visibleTodos]);
```

Comment:
Are you sure that you need `visibleTodos` in the dependency here?
```suggestion
  useEffect(() => {
    getTodosFromServer();
  }, []);
```

Reply:
Yes, it works incorrectly without it